code/datums/mutations.dm

```diff
@@ -4,66 +4,54 @@ GLOBAL_LIST_EMPTY(mutations_list)
-	if(!se_string || lentext(se_string) < DNA_STRUC_ENZYMES_BLOCKS * DNA_BLOCK_SIZE)
-		return
-	var/before = copytext(se_string, 1, ((dna_block - 1) * DNA_BLOCK_SIZE) + 1)
-	var/injection = num2hex(on ? rand(lowest_value, (256 * 16) - 1) : rand(0, lowest_value - 1), DNA_BLOCK_SIZE)
-	var/after = copytext(se_string, (dna_block * DNA_BLOCK_SIZE) + 1, 0)
-	return before + injection + after
-
-/datum/mutation/human/proc/set_block(mob/living/carbon/owner, on = 1)
-	if(owner && owner.has_dna())
-		owner.dna.struc_enzymes = set_se(owner.dna.struc_enzymes, on)
-
-/datum/mutation/human/proc/check_block_string(se_string)
-	if(!se_string || lentext(se_string) < DNA_STRUC_ENZYMES_BLOCKS * DNA_BLOCK_SIZE)
-		return 0
-	if(hex2num(getblock(se_string, dna_block)) >= lowest_value)
-		return 1
-
-/datum/mutation/human/proc/check_block(mob/living/carbon/human/owner, force_powers=0)
-	if(check_block_string(owner.dna.struc_enzymes))
-		if(prob(get_chance)||force_powers)
-			. = on_acquiring(owner)
-	else
-		. = on_losing(owner)
-
-/datum/mutation/human/proc/on_acquiring(mob/living/carbon/human/owner)
-	if(!owner || !istype(owner) || owner.stat == DEAD || (src in owner.dna.mutations))
+	var/datum/dna/dna
+	var/mob/living/carbon/human/owner
+	var/instability = 0 //instability the holder gets when the mutation is not native
+	var/blocks = 4 //Amount of those big blocks with gene sequences
+	var/difficulty = 8 //Amount of missing sequences. Sometimes it removes an entire pair for 2 points
+	var/timed = FALSE   //Boolean to easily check if we're going to self destruct
+	var/alias           //'Mutation #49', decided every round to get some form of distinction between undiscovered mutations
+	var/scrambled = FALSE //Wheter we can read it if it's active. To avoid cheesing with mutagen
+	var/class           //Decides player accesibility, sorta
+	//MUT_NORMAL - A mutation that can be activated and deactived with structural enzymes
+	//MUT_EXTRA - A mutation that is in the mutations tab, and can be given and taken away through though the DNA console. Has a 0 before it's name in the mutation section of the dna console
+	//MUT_OTHER Cannot be interacted with by players through normal means. I.E. wizards mutate
+
+/datum/mutation/human/New(class_ = MUT_OTHER, timer)
+	. = ..()
+	class = class_
+	if(timer)
+		addtimer(CALLBACK(src, .proc/remove), timer)
+		timed = TRUE
+
+/datum/mutation/human/proc/on_acquiring(mob/living/carbon/human/H)
+	if(!H || !istype(H) || H.stat == DEAD || (src in H.dna.mutations))
 		return TRUE
-	if(species_allowed.len && !species_allowed.Find(owner.dna.species.id))
+	if(species_allowed.len && !species_allowed.Find(H.dna.species.id))
 		return TRUE
-	if(health_req && owner.health < health_req)
+	if(health_req && H.health < health_req)
 		return TRUE
-	if(limb_req && !owner.get_bodypart(limb_req))
+	if(limb_req && !H.get_bodypart(limb_req))
 		return TRUE
-	owner.dna.mutations.Add(src)
+	owner = H
+	dna = H.dna
+	dna.mutations[src] = type
```

Comment:
What possible use this index has ? I don't see you using this anywhere and it doesn't make much sense.